packages/core/core/src/PackagerRunner.js

```diff
@@ -73,121 +84,84 @@ export default class PackagerRunner {
+    //  - not sure if that's a bad thing or not
+    let hashRefToId = {};
+    // skip inline bundles, they will be processed via the parent bundle
+    let bundles = bundleGraph.getBundles().filter(bundle => !bundle.isInline);
+    await Promise.all(
+      bundles.map(async bundle => {
+        let info = await this.processBundle(bundle, bundleGraph, ref);
+        hashRefToId[bundle.hashReference] = bundle.id;
+        bundleInfoMap[bundle.id] = info;
+      }),
+    );
+    let hashRefToNameHash = generateHashRefToNameHashMap(
+      bundles,
+      bundleInfoMap,
+      hashRefToId,
+    );
+    await Promise.all(
+      bundles.map(bundle =>
+        this.writeToDist({
+          bundle,
+          info: bundleInfoMap[bundle.id],
+          hashRefToNameHash,
+          bundleGraph,
         }),
-      );
-    }
-
-    await Promise.all(promises);
+      ),
+    );
     await dispose();
   }
 
-  async writeBundle(
+  async processBundle(
     bundle: InternalBundle,
     bundleGraph: InternalBundleGraph,
     bundleGraphReference: number,
-  ) {
+  ): Promise<{|...BundleInfo, cacheKey: string|}> {
     let start = Date.now();
 
     let cacheKey = await this.getCacheKey(bundle, bundleGraph);
-    let {size} =
-      (await this.writeBundleFromCache({bundle, bundleGraph, cacheKey})) ||
-      (await this.writeBundleFromWorker({
+    let {hash, hashReferences} =
+      (await this.getBundleInfoFromCache()) ??
+      (await this.getBundleInfoFromWorker({
         bundle,
-        cacheKey,
         bundleGraphReference,
+        cacheKey,
         options: this.options,
         config: this.config,
       }));
 
-    return {
-      time: Date.now() - start,
-      size,
-    };
+    return {time: Date.now() - start, hash, hashReferences, cacheKey};
   }
 
-  async writeBundleFromCache({
-    bundle,
-    bundleGraph,
-    cacheKey,
-  }: {|
-    bundle: InternalBundle,
-    bundleGraph: InternalBundleGraph,
-    cacheKey: string,
-  |}) {
-    if (this.options.disableCache) {
-      return;
-    }
-
-    let cacheResult = await this.readFromCache(cacheKey);
-    if (cacheResult == null) {
-      return;
-    }
-
-    let {contents, map} = cacheResult;
-    let {size} = await this.writeToDist({
-      bundle,
-      bundleGraph,
-      contents,
-      map,
-    });
-
-    return {size};
+  getBundleInfoFromCache() {
+    // TODO: implement
+    return null;
   }
 
-  async packageAndWriteBundle(
+  async getBundleInfo(
     bundle: InternalBundle,
     bundleGraph: InternalBundleGraph,
-    cacheKey: string,
+    cacheKey: ?string,
   ) {
-    let start = Date.now();
+    let {contents, map} = await this.getBundleResult(bundle, bundleGraph);
 
-    let {contents, map} = await this.getBundleResult(
-      bundle,
-      bundleGraph,
-      cacheKey,
-    );
-    let {size} = await this.writeToDist({
-      bundle,
-      bundleGraph,
-      contents,
-      map,
-    });
+    let info = {};
+    // ? Why would cacheKey be null
```

Comment:
Was this maybe for inline bundles?

Reply:
It was, not relevant anymore now that `getBundleInfo` is writing to cache instead of `getBundleResult`.